### DISCUSSION TITLE
제페토 카메라 코드로 조절하는 법 알 수 있을까요?

### DISCUSSION THREAD

**Discussion by hipsc159** [naverz/zepeto-studio-kor#370] [5. Q&A]
특정한 장소로 들어가면 카메라 zoom거리나 offset을 변경하고 싶은데, 찾기 어렵네요..

**Comment by junekinns** [comment]
저는 ZepetoPlayers 게임오브젝트에 별도의 ZepetoScript 컴포넌트를 추가하여 해당 스크립트에서 카메라를 가져오는 방법으로 제어했습니다.
다만 `FindChild("")`는 왜인지모르겠지만 동작하지 않아서, 어쩔수없이 `GetChild(index)`로 가져왔네요. `GetChild(2)`인 이유는 게임오브젝트 하이어라키순서 때문입니다.
예를들어,  카메라를 Orthographic으로 바꾸고, 사이즈를 조절하는 스크립트는 다음과 같이 하면 됩니다.
더 좋은 방법이 있을지도 모르지만 일단 이렇게 하면 잘됩니다. :)
  

```
import { ZepetoScriptBehaviour } from 'ZEPETO.Script'
import { GameObject, Object, Transform, Camera } from "UnityEngine";
export default class Cam extends ZepetoScriptBehaviour {
    public camSize : int;
    private cam: Camera;
    Start() {    
       var zepetoCameraTr = this.gameObject.transform.GetChild(2); //FindChild("") not working? 
       var cameraParent = zepetoCameraTr.GetChild(0);
       var zepetoCamera = cameraParent.GetChild(0);
       this.cam = zepetoCamera.GetComponent<Camera>();
       this.cam.orthographic = true;
    }
     
    Update(){
       this.cam.orthographicSize = this.camSize;
    }

}
```

**Reply by hipsc159** [reply]
넵 답변 정말 감사합니다.
하지만 저는 zepeto players에서 카메라 조절하는게 궁금합니다!

![Screenshot_4](https://user-images.githubusercontent.com/68143858/153704394-3d886f93-974b-4d58-a210-ae18da0d59c5.png)

**Comment by pumfit** [comment]
ZepetoPlayers instance에서 아래 코드와 같이 zoom, offset 값 변경이 가능합니다.
또한 코드에서 카메라를 가져와 직접 제어할때 아래와 같이 접근도 가능합니다!
>    
    Start() {       
        //camera 접근 방법
        let camera = ZepetoPlayers.instance.ZepetoCamera.camera;

        // Zoom 변경 함수
        ZepetoPlayers.instance.ZepetoCamera.DoZoom(2);

        //zoom 최대,최소값 조절
        ZepetoPlayers.instance.cameraData.maxZoomDistance = 8;
        ZepetoPlayers.instance.cameraData.minZoomDistance = 1;

        //offset 변경
        ZepetoPlayers.instance.cameraData.offset = new Vector3(0.1,0.2,0.3);
    }

![image](https://user-images.githubusercontent.com/46295539/153737261-a15f5b4e-1b6d-43d5-aecd-d73e5a49fadd.png)

**Reply by junekinns** [reply]
감사합니다 👍🏻

**Reply by hipsc159** [reply]
정말 감사합니다!! 완벽하게 작동합니다.
궁금한 점이 있는데 ,

1. let camera = ZepetoPlayers.instance.ZepetoCamera.camera;
2. ZepetoPlayers.instance.ZepetoCamera.DoZoom(2);

혹시 1. 카메라 변수를 할당받는 것과 2. DoZoom 함수 (zoom 변경 함수)는 어떤 역할을 하는지 알 수 있을까요??
1,2 을 하지 않아도 기능은 작동하기에 질문드립니다!

**Reply by pumfit** [reply]
1번은 위 답변에서 카메라를 가져오는 방식을 알려주셔서 위 방식 말고도 접근 가능하다는 것을 알려 드리려고 적었고 2번은 특정 장소에서 들어갈때 zoom거리를 변경하신다기에 해당 함수를 이용해 zoom값을 바로 변경할 수 있다는 걸 알려드리려 했습니다! 답변을 다시 보니 제 설명이 부족했네요ㅎㅎ 잘 작동되신다니 다행입니다!😃

**Reply by hipsc159** [reply]
아~ 이해했습니다. 신경써주셔서 감사합니다 :)
좋은 하루 보내세요~~!
